src/object.js

```diff
@@ -32,15 +32,24 @@ var createTestDoublesForFunctionNames = (names) =>
 
 var createTestDoubleViaProxy = (name, config) => {
   ensureProxySupport(name)
-  const obj = {}
-  return new Proxy(obj, {
-    get (target, propKey, receiver) {
-      if (!obj.hasOwnProperty(propKey) && !_.includes(config.excludeMethods, propKey)) {
-        obj[propKey] = tdFunction(`${nameOf(name)}.${String(propKey)}`)
-      }
-      return obj[propKey]
-    }
+
+  const generateHandler = (internalName) => ({
+    get: (target, propKey) => generateGet(target, propKey, internalName)
   })
+
+  const generateGet = (target, propKey, internalName) => {
+    if (!target.hasOwnProperty(propKey) && !_.includes(config.excludeMethods, propKey)) {
+      let nameWithProp = `${nameOf(internalName)}.${String(propKey)}`
+      const obj = tdFunction(nameWithProp)
+      obj.___mocks___ = {}
```

Comment:
I'm just now getting around to reviewing this, @lgandecki. Sorry for being slow. The only red flag I see is this `___mocks___` scratchpad. So far the library has taken pretty extreme pains to avoid keeping track of _anything_ on the fakes it creates, instead storing them in memory in a private way that isn't exposed to the user. 

I like this feature, but I wouldn't want to break from that tradition here if we can avoid it. I don't have an immediate solution, but if you check any code described as a "store" elsewhere that's how we've been doing. Most important thing is that whatever it is be registered with `td.reset` to reset appropriately between tests.

Reply:
Good Point! There was actually no need for this scratchpad anymore, it was a leftover of a previous implementation.. sorry! I just removed it.

I also added a test that verifies that td.reset does work properly here.

Please let me know if there is something more you would like to have changed here.